lib/resty/mlcache.lua

```diff
@@ -513,7 +529,12 @@ function _M:get(key, opts, cb, ...)
 
     -- set our own worker's LRU cache
 
-    set_lru(self, key, data, ttl, neg_ttl)
+    data, err = set_lru(self, key, data, ttl, neg_ttl)
+    if data == CACHE_MISS_SENTINEL_LRU then
```

Comment:
Maybe we cab simply ignore the first return value of `set_lru` here? Except when it is `nil`, which indicates an error returned by the user (`nil + err`). If you'd rather test the second return value for that, that's fine too imho.

Reply:
No, we must use the `set_lru` result: if a `lru_callback` has been set, data might be transformed, so the result is not necessarily the same as the `data` parameter.

https://github.com/thibaultcha/lua-resty-mlcache/pull/29/files/84bdb540283d2f3f439c7fdaa0b75b29ccc2838b#diff-1d36902a4b814b8c38955c37ed51f2c4R280

Reply:
Right. Forgive me for I have not touched this library in a while... It is a bit blurry :) I was trying to avoid the extra branch and potential future programming errors now that we can modify `data` on the fly, and need to re-evaluate `== CACHE_MISS_SENTINEL_LRU`. I was hoping maybe we could avoid it, but it's fine as is for now at least 👍